### PR TITLE
Blobstore image upload with recommendation form MVP

### PR DESCRIPTION
**Pull request by Sherry2001**
In the second recommendation form, you are now able to submit a picture along with your recommendation. 

deployed version: http://shershi-step-2020.appspot.com/
html preview: https://htmlpreview.github.io/?https://github.com/Sherry2001/step/blob/master/portfolio/src/main/webapp/index.html
**For some reason uploading file form doesn't work in the deployed version, giving a status 500 error, though it works locally. Here's the error in the logger:

Screenshot 2020-06-09 at 9 30 35 PM

### CODE REVIEW

File: portfolio/src/main/webapp/script.js [hunk 1]
```diff
@@ -55,6 +55,13 @@ function createRecommendationElement(toDo) {
   comment.innerHTML = toDo.comment; 
   recommendation.appendChild(comment);
 
+  if (toDo.imageUrl !== undefined){
```
Comment (ricazhang) [inline portfolio/src/main/webapp/script.js:58]: `if (!toDo.imageUrl) {` ?

File: portfolio/src/main/webapp/script.js [hunk 2]
```diff
@@ -14,7 +14,7 @@ function getData(maxLoad) {
     const musicList = document.getElementById('music-list');
     const movieList = document.getElementById('movie-list');
     const travelList = document.getElementById('travel-list');
-
+
```
Comment (ricazhang) [inline portfolio/src/main/webapp/script.js:17]: Revert this change?

File: portfolio/src/main/webapp/script.js [hunk 3]
```diff
@@ -137,14 +166,16 @@ document.addEventListener('DOMContentLoaded', (event) => {
...
     .then(() => {
-        fetch (url, {method: 'POST'});
-    }).then(() => {
-        getData();
-        respond("Thank you for your recommendation :)");
+      fetch (url, {method: 'POST'});
     })
+    .then(() => {
+      getData();
+      respond("Thank you for your recommendation :)");
```
Comment (ricazhang) [inline portfolio/src/main/webapp/script.js:176]: single quotes

File: portfolio/src/main/webapp/script.js [hunk 4]
```diff
@@ -137,14 +166,16 @@ document.addEventListener('DOMContentLoaded', (event) => {
     const url = '/data?name=' + name + '&category=' + category + '&content=' + recommendation +
                '&comment=' + comment;
 
+
```
Comment (ricazhang) [inline portfolio/src/main/webapp/script.js:169]: extra newline here and line 148

File: portfolio/src/main/webapp/script.js [hunk 5]
```diff
@@ -137,14 +166,16 @@ document.addEventListener('DOMContentLoaded', (event) => {
...
-    }).then(() => {
-        getData();
-        respond("Thank you for your recommendation :)");
+      fetch (url, {method: 'POST'});
     })
+    .then(() => {
+      getData();
+      respond("Thank you for your recommendation :)");
+    })
+    .catch(error => respond('Uh oh, error: ' + error.message))
```
Comment (ricazhang) [inline portfolio/src/main/webapp/script.js:178]: Does this need surrounding parentheses so that it has access to the respond method?
`catch((error) => respond(...)`

**Review by ricazhang — APPROVED** [review]

File: portfolio/src/main/webapp/script.js (hunk 1 above)
Comment (Sherry2001) [reply portfolio/src/main/webapp/script.js:58]: ah.. just tried- that doesn't work. I think because the value of toDo.imageUrl in datastore is = undefined?

Comment (ricazhang) [reply portfolio/src/main/webapp/script.js:58]: Oops sorry I meant
`if (toDo.imageUrl)`

File: portfolio/src/main/webapp/script.js (hunk 2 above)
Comment (Sherry2001) [reply portfolio/src/main/webapp/script.js:17]: okay, I still kept an empty line there.

File: portfolio/src/main/webapp/script.js (hunk 4 above)
Comment (Sherry2001) [reply portfolio/src/main/webapp/script.js:169]: okay, deleted!

File: portfolio/src/main/webapp/script.js (hunk 3 above)
Comment (Sherry2001) [reply portfolio/src/main/webapp/script.js:176]: yes!

File: portfolio/src/main/webapp/script.js (hunk 5 above)
Comment (Sherry2001) [reply portfolio/src/main/webapp/script.js:178]: oh i think if there's only one parameter it should work? but I never tested it. I added the parenthesis anyway!

File: portfolio/src/main/webapp/script.js (hunk 1 above)
Comment (Sherry2001) [reply portfolio/src/main/webapp/script.js:58]: oh right! makes sense!

File: portfolio/src/main/webapp/script.js (hunk 5 above)
Comment (ricazhang) [reply portfolio/src/main/webapp/script.js]: Oh yeah you're right. While I was looking this up, I found in the [eslint guidelines](https://eslint.org/docs/rules/arrow-parens) that they suggest always using parentheses to eliminate ambiguity. I think you don't need to follow that here since you aren't using that linter, but just an FYI!

Comment (Sherry2001) [reply portfolio/src/main/webapp/script.js]: ah okay, will remember!

**Review by Alfredhgoog — APPROVED** [review]
